airflow/ui/src/pages/Variables/AddVariableForm.tsx

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, HStack, Input, Spacer, Text } from "@chakra-ui/react";
+import { useEffect, useMemo } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { FiSave } from "react-icons/fi";
+
+import { ErrorAlert } from "src/components/ErrorAlert";
+import { Button } from "src/components/ui";
+import { useAddVariable } from "src/queries/useAddVariable";
+
+export type AddVariableBody = {
+  description: string | undefined;
+  key: string;
+  value: string;
+};
+
+type AddVariableFormProps = {
+  onClose: () => void;
+};
+
+const AddVariableForm: React.FC<AddVariableFormProps> = ({ onClose }) => {
+  const { addVariable, error, isPending } = useAddVariable(onClose);
+  const addVariableParams: AddVariableBody = useMemo(
+    () => ({
+      description: undefined,
+      key: "",
+      value: "",
+    }),
+    [],
+  );
+
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+    watch,
+  } = useForm<AddVariableBody>({
+    defaultValues: addVariableParams,
+  });
+
+  const { key, value } = watch();
+
+  useEffect(() => {
+    reset(addVariableParams);
+  }, [addVariableParams, reset]);
+
+  const onSubmit = (data: AddVariableBody) => {
+    addVariable(data);
+  };
+
+  return (
+    <>
+      <ErrorAlert error={error} />
+      <Text fontSize="md" mb={2} mt={4}>
+        Key <span style={{ color: "red" }}>*</span>
+      </Text>
+      <Controller
+        control={control}
+        name="key"
+        render={({ field }) => <Input {...field} required size="sm" />}
```

Comment:
The DB schema is limiting the volume, if I add data longer than the DB can hold, an internal server error is raised.

![image](https://github.com/user-attachments/assets/31ee9314-563a-4e07-8b28-a0613df76a67)

Can you limit the entry boxes to the DB limits?
And is there abetter error handling in FastAPI needed (==such that not internal server error is returned, better in a different PR) or is it not correctly handled in the form?

Reply:
For now, I will have a client side check about it.

We can sure raise a issue to fix the internal error message on data overload.

Reply:
Yes, low hanging fruit would be to limit the entry box length. I assume this is sufficient. If I "hack" the form and then recive an internal server error... then anyway my fault.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength

Reply:
the server should respond a proper 422. (Client error) we need to improve that

Reply:
I assume this is also already resolved by other OR, correct?